### PR TITLE
Update pre-requisites for native aot publishing

### PR DESCRIPTION
**Pull request by jkotas**
## Summary

Trim to list pre-requisites for native aot publishing to just what is actually required.

Addressed feedback from https://github.com/dotnet/runtime/issues/69739#issuecomment-1264133462

### CODE REVIEW

**Comment by jkotas** [comment]
cc @am11

File: docs/core/deploying/native-aot/index.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ On Linux, install clang and developer packages for libraries that .NET runtime d
 - Alpine (3.15+)
 
     ```sh
-    sudo apk add clang gcc lld musl-dev build-base zlib-dev
+    sudo apk add build-base zlib-dev
```
Comment (am11) [inline docs/core/deploying/native-aot/index.md:38]: Line 27 notes `install clang and developer packages`, but `build-base` on Alpine (and `build-essential` on Debian/Ubuntu) is a meta package which include gcc toolchain but not llvm-toolchain. Should we update the note to `install compiler toolchain and developer packages`.

Alternatively:

```suggestion
    sudo apk add clang build-base zlib-dev
```

Comment (am11) [inline docs/core/deploying/native-aot/index.md:38]: Line 27 notes `install clang and developer packages`, but `build-base` on Alpine (and `build-essential` on Debian/Ubuntu) is a meta package which includes gcc-toolchain but not llvm-toolchain. Should we update the note to `install compiler toolchain and developer packages`.

Alternatively:

```suggestion
    sudo apk add clang build-base zlib-dev
```

Comment (jkotas) [reply docs/core/deploying/native-aot/index.md]: I have done both. We are doing our testing with clang, so it is what we should tell people to use. If we were to omit clang here, we would need to redo how we do testing.

**Review by am11 — APPROVED** [review]
LGTM, thanks!

**Review by gewarren — APPROVED** [review]

**Pull request merged**